### PR TITLE
sftpgo: init at 2.4.0

### PR DESCRIPTION
**Pull request by thenonameguy**
###### Description of changes

New package added, this adds capability to nixpkgs to run rootless SFTP servers easily.

https://github.com/drakkan/sftpgo

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thenonameguy** [comment]
TODO:
- [ ] add shell completions via `sftpgo gen completion $SHELL`
- [ ] add man pages via `sftpgo gen man`

**Comment by rapenne-s** [comment]
hi,
the commit message doesn't fit [CONTRIBUTING](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes)

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -13538,6 +13538,12 @@
     githubId = 3105057;
     name = "Jan Beinke";
   };
+  thenonameguy = {
+    email = "thenonameguy24@gmail.com";
+    name = "Krisztian Szabo";
+    github = "thenonameguy";
+    githubId = 2217181;
+  };
```
Comment (marsam) [inline maintainers/maintainer-list.nix:13546]: This change needs to be in a separated commit. 
Besides that, LGTM

Comment (thenonameguy) [reply maintainers/maintainer-list.nix:13546]: Thanks for the review! Addressing...

Comment (thenonameguy) [reply maintainers/maintainer-list.nix:13546]: Ah I guess the maintainer list additon commit has to come first. Fixing.

**Comment by thenonameguy** [comment]
@marsan ready for review :bow:

**Review by marsam — APPROVED** [review]

**Pull request merged**